Add Description-ContextMenu Entry to bSDD Properties

c-mellueh opened the pull request:
I like the "IFC Description" Context Menu. So i added the same functionality for bSDD Properties.
<img width="457" height="583" alt="image" src="https://github.com/user-attachments/assets/c0fe6342-d6f6-4f57-b034-7da77e9e712b" />


<img width="721" height="309" alt="image" src="https://github.com/user-attachments/assets/57ab1e26-c051-4433-85d7-3b92b3466ef4" />

c-mellueh committed:
fix property counting

c-mellueh committed:
add bSDD Description context menu

Andrej730 commented:
@c-mellueh 

Thanks for the PR, looks good in general.

A few things I've noticed:

1. Can you please submit bsdd tests fixes in a separate PR as those seem unrelated?
2. `op_description: BIM_OT_show_bsdd_description` - this is incorrect, it's a wrong type, `layout.operator` returns `OperatorProperties`. I think it would be better to annotate `bsdd_property` as `BSDDProperty` instead.
3. `# pset_name = result.get("propertySet") or "undefined"` - left out commented code.
4. `self.bl_label = result.get("name") or "undefined"` - dead code.
5. `total_results = 0` - dead code.
6. I think there should be some separation in case if there are both description and definition present:

<img width="670" height="444" alt="image" src="https://github.com/user-attachments/assets/ddfdee18-c67a-4286-a683-4890bfbcb2ed" />

c-mellueh committed:
remove dead code

c-mellueh committed:
fix typing error

c-mellueh committed:
add linebreak wich dashed line between description an definition

c-mellueh committed:
reformatted to fit black

c-mellueh commented:
@Andrej730  thank you for your feedback. 
I implemented your requests.
 Check out [5fa5e18](https://github.com/IfcOpenShell/IfcOpenShell/pull/6970/commits/5fa5e189cac007cc03f1d73932d6bcc21f9e49f7) if you're happy with my solution of separating description and definition.
I added a dashed line:
<img width="667" height="451" alt="Screenshot_20250804_130552" src="https://github.com/user-attachments/assets/e0bf144e-e00f-4f43-a6b2-7708c4782b3d" />

Andrej730 commented:
Looks good, thank you!
